### PR TITLE
UCR ES transforms

### PR DESCRIPTION
**Pull request by emord**
@esoergel This is what you reported in your PR a month or so ago.

Also caught this coming up in the laboratory experiments for enikshay

buddy @benrudolph

### CODE REVIEW

**Comment by emord** [comment]
Relevant comment/bug report: https://github.com/dimagi/commcare-hq/pull/14183#issuecomment-266596018

File: corehq/apps/userreports/es/data_source.py [hunk 1]
```diff
@@ -123,16 +128,24 @@ def _parse_buckets(buckets, remaining_agg_columns, current_bucket_name, past_buc
...
 
-        ret = []
+        ret = OrderedDict()
         start = start or 0
         end = start + (limit or len(hits))
         relevant_hits = hits[start:end]
 
         for row in relevant_hits:
-            r = {}
+            r_ = {}
```
Comment (esoergel) [inline corehq/apps/userreports/es/data_source.py]: out of curiosity, what's with the trailing underscore?

**Review by esoergel — APPROVED** [review]
That parsing code is wicked complicated, but this all looks good to me.

File: corehq/apps/userreports/es/data_source.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/userreports/es/data_source.py]: (offlined, it was so `pdb` works)

**Comment by emord** [comment]
> That parsing code is wicked complicated, but this all looks good to me.

I bet sqlalchemy does some pretty complicated stuff too :)

It's likely that a lot of this could be pushed into the reporting code and as an `ESReport` class, like the `SqlReport` class. Then the UCR could inherit from that and it could be used in othe reports

**Pull request merged**